src/python/WMCore/MicroService/Unified/MSOutput.py

```diff
@@ -284,41 +284,80 @@ def makeSubscriptions(self, workflow):
+                                self.logger.warning(msg)
+                                continue
+                            ddmReqList.append(ddmRequest)
+                    else:
+                        # FIXME:
+                        #    We need to create the campaignMap and use it for
+                        #    creating the requests for the nonRelVal workflows
+                        #    it should be also the case when we migrate to Rucio
                         ddmRequest = DDMReqTemplate('copy',
                                                     item=workflow['OutputDatasets'],
                                                     n=workflow['numberOfCopies'],
                                                     site=workflow['destination'])
-                    except Exception as ex:
-                        msg = "Could not create DDMReq for Workflow: {}".format(workflow['RequestName'])
-                        msg += "Error: {}".format(ex)
-                        self.logger.exception(msg)
-                        return workflow
-
-                    try:
-                        # In the message bellow we may want to put the list of datasets too
-                        msg = "Making transfer subscriptions for {}".format(workflow['RequestName'])
-                        self.logger.info(msg)
-                        ddmResult = self.ddm.makeRequest(ddmRequest)
-                    except Exception as ex:
-                        msg = "Could not make transfer subscription for Workflow: {}".format(workflow['RequestName'])
-                        msg += "Error: {}".format(ex)
-                        self.logger.exception(msg)
-                        return workflow
+                        ddmReqList.append(ddmRequest)
+                except Exception as ex:
+                    msg = "Could not create DDMReq for Workflow: {}".format(workflow['RequestName'])
+                    msg += "Error: {}".format(ex)
+                    self.logger.exception(msg)
+                    return workflow
+
+                try:
+                    # In the message bellow we may want to put the list of datasets too
+                    msg = "Making transfer subscriptions for {}".format(workflow['RequestName'])
+                    self.logger.info(msg)
+                    ddmResultList = self.ddm.makeAggRequests(ddmReqList, aggKey='item')
+                except Exception as ex:
+                    msg = "Could not make transfer subscription for Workflow: {}".format(workflow['RequestName'])
+                    msg += "Error: {}".format(ex)
+                    self.logger.exception(msg)
+                    return workflow
 
+                ddmStatusList = ['new', 'activated', 'completed', 'rejected', 'cancelled']
+                transferIDs = []
+                transferStatusList = []
+                for ddmResult in ddmResultList:
                     if 'data' in ddmResult.keys():
```

Comment:
Will a DDM http request always return the `data` key?
I still wonder if we could hit the same issue as I reported above, some tests going through, some not, and we marking the workflow as done.

Reply:
If the request is actually done the 'data' field will be there (unless DDM changes its interface to the outer world). Here is an example [1]. And if in case of connectivity problems or something like that, the response is incomplete and we fail to fetch the 'data' - even then we cannot mark the Workflow as done because it will fail to accumulate the statuses few lines bellow, so the Workflow will be retired on the next cycle.


[1]
```
{u'data': [{u'dn': u'/DC=ch/DC=cern/OU=computers/CN=wmagent/vocms0192.cern.ch',
            u'first_request': u'2020-05-25T14:20:15 UTC',
            u'group': u'DataOps',
            u'item': [u'/LQLQToTopMuTopTau_M-1200_TuneCP5_13TeV_pythia8/RunIIFall17NanoAODv5-PU2017_12Apr2018_Nano1June2019_102X_mc2017_realistic_v7-v1/NANOAODSIM'],
            u'last_request': u'2020-05-25T14:20:33 UTC',
            u'n': 1,
            u'request_count': 2,
            u'request_id': 47373,
            u'site': [u'T1_UK_RAL_Disk',
                      u'T2_PL_Swierk',
                      u'T2_RU_INR',
                      u'T2_BE_IIHE',
                      u'T2_US_MIT',
...
                      u'T1_IT_CNAF_Disk'],
            u'status': u'new',
            u'user': u'dmwm_alandev@cmsweb.cern.ch'}],
 u'message': u'',
 u'result': u'OK'}
```

Reply:
Anyway I just added few  additional checks.